es6/docxtemplater.js

```diff
@@ -28,15 +28,22 @@ const {
 const currentModuleApiVersion = [3, 21, 0];
 
 const Docxtemplater = class Docxtemplater {
-	constructor() {
-		if (arguments.length > 0) {
+	constructor(zip, { modules = [] } = {}) {
+		if (!Array.isArray(modules)) {
 			throw new Error(
-				"The constructor with parameters has been removed in docxtemplater 3, please check the upgrade guide."
+				"Please pass modules as an array"
 			);
 		}
 		this.compiled = {};
 		this.modules = [commonModule()];
 		this.setOptions({});
+		modules.forEach(module => {
+			this.attachModule(module);
+		});
+		if (zip) {
+			this.loadZip(zip);
```

Comment:
You can actually chain the commands, eg write : this.loadZip(zip).compile();

Reply:
Yes, I guess it's better to use the `builder pattern` if its available. 😊